micrometer-tracing-bridges/micrometer-tracing-bridge-brave/src/test/java/io/micrometer/tracing/brave/bridge/BaggageTests.java

```diff
@@ -258,6 +273,26 @@ void baggageTagKey() {
         then(mutableSpan.tags().get(TAG_KEY)).isEqualTo(TAG_VALUE);
     }
 
+    @Test
+    void baggageWithObservationApiWithRemoteFields() {
+        Observation observation = Observation.start("foo", observationRegistry)
+            .lowCardinalityKeyValue(KEY_1, TAG_VALUE)
+            .highCardinalityKeyValue(OBSERVATION_BAGGAGE_KEY, OBSERVATION_BAGGAGE_VALUE);
+        try (Scope scope = observation.openScope()) {
+            then(tracer.getBaggage(KEY_1).get()).isEqualTo(TAG_VALUE);
+            then(tracer.getBaggage(OBSERVATION_BAGGAGE_KEY).get()).isEqualTo(OBSERVATION_BAGGAGE_VALUE);
+        }
+        then(tracer.currentSpan()).isNull();
+        then(tracer.getBaggage(KEY_1).get()).isNull();
+        then(tracer.getBaggage(OBSERVATION_BAGGAGE_KEY).get()).isNull();
```

Comment:
Can we do these check before opening the scope too?

Reply:
Of course